components/core/src/GlobalMySQLMetadataDB.cpp

```diff
@@ -148,10 +148,10 @@ void GlobalMySQLMetadataDB::update_metadata_for_files (const std::string& archiv
     }
     auto& statement_bindings = m_upsert_file_statement->get_statement_bindings();
     for (auto file : files) {
-        const auto& id_as_string = file->get_id_as_string();
+        const auto id_as_string = file->get_id_as_string();
```

Comment:
In `MetadataDB::update_files` and `GlobalSQLiteMetadataDB::update_metadata_for_files`, I wrote calls like:

```
m_upsert_file_statement->bind_text(enum_to_underlying_type(FilesTableFieldIndexes::OrigFileId) + 1, file->get_orig_file_id_as_string(), false);
```

Passing `false` means that the method does not need to copy the string; instead, it expects it to live until the statement is run (I probably should've added a comment to make that clear). With the change in return type of `get_id_as_string` and `get_orig_file_id_as_string`, the lifetime of the string is no longer clear. So, just as you do here, I would save the return values in local variables in those two methods.